New version v1.2.0

TheKeymaster opened the pull request:
Close #29 
Close #30 

Fix code styling from index and using latest libflexport version.
Also minor changes in files were done.

**!!composer.json refers to `require-dev`, please change to libflexport `v1.0.2` before merging!!**

Commit:
Fix code styling from index and using latest libflexport version. composer.json refers to require-dev, please change before merging. Also minor changes were done.

TheKeymaster commented:
@georgms you already can review. I just need to change the composer.json before merging.

Commit:
edit composer.json for correct v1.0.2 libflexport match

TheKeymaster commented:
@georgms pls review. Thx

georgms commented on DokuwikiXMLExport.php:
```diff
@@ -85,9 +85,9 @@ private function getPageIds()
+        // Get all pages that do have a description and a title set
         $pagesAndDeletedPages = array_filter($pagesAndDeletedPages, function ($page, $k) {
-            return (p_get_metadata($page)['description'] !== '' && p_get_metadata($page)['title'] !== '');
+            return (p_get_metadata($page)['description'] !== '' && !empty(p_get_metadata($page)['title']));
```
Explain why you are checking description for `''` and title with `empty`. Probably extract both conditions into their own variables to make the statement clearer.

TheKeymaster replied:
Actually both should be tested for empty. Will add.

georgms commented on _test/page.getter.test.php:
```diff
@@ -48,7 +48,7 @@ public function test_get_pages_without_title_gets_pages_sorted_by_time()
         foreach ($pagesHaveNoTitle as $pageHasNoTitle) {
             Helper::savePages([$pageHasNoTitle], false);
-            sleep(60);
+            sleep(1);
```
That's a 60x performance increase. Nice 😃

TheKeymaster replied:
It is ;) :racehorse: :white_check_mark:

georgms commented on index.php:
```diff
@@ -7,37 +7,69 @@
-} else {
-    $count = 20;
-}
+const DOKUWIKI_INC = 'DOKU_INC';
```
Always add doc block for consts.

georgms commented on index.php:
```diff
@@ -7,37 +7,69 @@
+function getUrlParam($paramName, $defaultValue)
+{
+    if (isset($_GET[$paramName])) {
+        return htmlspecialchars($_GET[$paramName]);
```
I'm not sure that escaping should be the responsibility of this method, but the docblock should state that and how the output is escaped.

georgms commented on index.php:
```diff
@@ -7,37 +7,69 @@
+ *
+ * @param $paramName string Name of the URL parameter
+ * @param $defaultValue string Default value if _GET parameter is not set
+ * @return string _GET parameter or default value of _GET parameter is not set
```
Is it really returning the `_GET parameter`?

TheKeymaster replied:
value of _GET param

georgms commented on index.php:
```diff
@@ -7,37 +7,69 @@
-}
-// Get $conf from DokuWiki for constructor
-global $conf;
+require_once(__DIR__ . '/DokuwikiXMLExport.php');
```
I won't require that for passing the review, but if you like you can wrap all the stuff (consts, methods) in a small class; classes are always good for maintainability and they don't cost nothin'

TheKeymaster replied:
Will implement this. It only makes sense.

georgms reviewed (changes requested):
Some minor changes

Commit:
fix review issues

TheKeymaster commented:
@georgms I fixed errything. Please check

georgms reviewed (changes requested):
Please fix merge conflicts.

Commit:
Merge branch 'dev_version_1_2_0'

Commit:
Merge branch 'master' of github.com:findologic/dokuwiki-plugin-findologic-xml-export

Commit:
fix

Commit:
add codeclimate.yml

TheKeymaster commented:
@georgms I fixed those merge conflicts so far. Could you approve my changes?

Commit:
remove file for next commit

Commit:
add example image

TheKeymaster commented:
Just included some folders that were missing. Please review this :) @georgms

georgms committed:
Merge remote-tracking branch 'origin/fix-filestructure' into dev_version_1_2_0

georgms commented on DokuwikiXMLExport.php:
```diff
@@ -72,14 +84,20 @@ private function getPageIds()
+        // Get all pages that do have a description and a title set
         $pagesAndDeletedPages = array_filter($pagesAndDeletedPages, function ($page, $k) {
-            return (p_get_metadata($page)['description'] !== '' && p_get_metadata($page)['title'] !== '');
+            $pageDescription = p_get_metadata($page)['description'];
```
I'd rather turn the variables into booleans, eg.

```php
$pageDescriptionIsNotEmpty = !empty(p_get_metadata($page)['description']);
$pageTitleIsNotEmpty = !empty(p_get_metadata($page)['title']);
return $pageDescriptionIsNotEmpty && $pageTitleIsNotEmpty;
```
(Hope I got the logic right)

This makes it easier to read IMHO.

TheKeymaster replied:
I thought about this but I wasn't sure if it makes sense to save the booleans in their variables. Will implement, because the readability is so much better.

TheKeymaster replied:
:white_check_mark:

georgms commented on DokuwikiXMLExport.php:
```diff
@@ -100,43 +119,54 @@ private function splitConfigToArray($config)
         $exporter = Exporter::create(Exporter::TYPE_XML, $count);
+
         $total = count($this->pages);
+        $submittedCount = $count;
```
Probably more clear if you rename the method argument from `$count` to `$submittedCount` and then do `$count = min($total, $submittedCount);`

TheKeymaster replied:
Agreed

TheKeymaster replied:
:white_check_mark:

georgms reviewed (changes requested):
* I've rebased this PR on `fix-filestructure` to make it easier to review.
* Please make sure that I didn't miss anything while merging.
* Some small changes, looking good otherwise.

Commit:
fix review issues

TheKeymaster commented:
> I've rebased this PR on fix-filestructure to make it easier to review.

That is very cool. Thanks for that Grand Master!

> Please make sure that I didn't miss anything while merging.

I do it very detailed, so we do not miss anything.

.
├── admin.php :white_check_mark: LGTM
├── admin.svg :white_check_mark: was not changed in this PR
├── composer.json :white_check_mark: LGTM
├── composer.lock :ballot_box_with_check: - excluded by `.gitignore`
├── conf
│   ├── default.php :white_check_mark: was not changed in this PR
│   └── metadata.php :white_check_mark: was not changed in this PR
├── DokuwikiPage.php :white_check_mark: LGTM
├── DokuwikiXMLExport.php :white_check_mark: LGTM
├── examples
│   └── example_admin.png :white_check_mark: was not changed in this PR
├── index.php :white_check_mark: LGTM
├── lang
│   ├── de
│   │   ├── lang.php :white_check_mark: LGTM
│   │   └── settings.php :white_check_mark: LGTM
│   └── en
│       ├── lang.php :white_check_mark: LGTM
│       └── settings.php :white_check_mark: LGTM
├── PageGetter.php :white_check_mark: LGTM
├── plugin.info.txt :white_check_mark: was not changed in this PR
├── README.md :white_check_mark: was not changed in this PR
├── resources
│   ├── edit.svg :white_check_mark: was not changed in this PR
│   ├── script.js :white_check_mark: was not changed in this PR
│   └── style.css :white_check_mark: was not changed in this PR
├── _test
│   ├── configuration.test.php :white_check_mark: LGTM
│   ├── export.call.test.php :white_check_mark: LGTM
│   ├── export.response.test.php :white_check_mark: LGTM
│   ├── general.test.php :white_check_mark: LGTM
│   ├── Helper.php :white_check_mark: LGTM
│   ├── page.getter.test.php :white_check_mark: LGTM
│   └── template.response.test.php :white_check_mark: LGTM
├── tpl
⠀⠀└── admin.twig :white_check_mark: LGTM

> Some small changes, looking good otherwise.

:white_check_mark: created commit where I fixed those changes.

TheKeymaster commented:
@georgms should we include the updated index.php with the OutputXMLHelper.php?

From:

 * index.php: https://github.com/findologic/dokuwiki-plugin-findologic-xml-export/pull/32/commits/d5f2a21771e57b2abc2fc52281c7f5c6237560bd#diff-828e0013b8f3bc1bb22b4f57172b019d
 * OutputXMLHelper.php: https://github.com/findologic/dokuwiki-plugin-findologic-xml-export/pull/32/commits/e4fe59b98f2dcc778352abff7451a3d279e5584d#diff-d1ccf947f8cfcf82ce9877d5d6c0f9e3

Please tell me, so we can finally merge this PR. :)

Commit:
New index.php version

georgms commented on OutputXMLHelper.php:
```diff
@@ -0,0 +1,93 @@
+     */
+    public function paramsValid($start, $count)
+    {
+        return (is_numeric($count) && is_numeric($start) && $start >= 0 && $count > 0);
```
Should use `is_int()` + test

TheKeymaster replied:
:white_check_mark: (tests are missing; will be implemented now....)

TheKeymaster replied:
:white_check_mark:

georgms commented on OutputXMLHelper.php:
```diff
@@ -0,0 +1,93 @@
+     * @param $defaultValue string Default value if _GET parameter is not set
+     * @return string value of the _GET parameter or default value if _GET parameter is not set
+     */
+    public function getUrlParam($paramName, $defaultValue)
```
Should pass `$_GET` as method argument in order to avoid using superglobals inside the method (not testable etc.)

TheKeymaster replied:
:white_check_mark:

georgms reviewed (changes requested):
Needs minor changes

Commit:
Parameters are now being checked for is_int instead for is_numeric

Commit:
Fix review issues

Commit:
Fix tests.

TheKeymaster commented:
@georgms could you review this?

georgms reviewed (approved):
LGTM
